anomaly_detector/anomaly_detector.py

```diff
@@ -163,55 +182,67 @@ def infer(self):
             now = datetime.datetime.now()
 
             # Get data for inference
-            data, json_logs = self._load_data(self.config.INFER_TIME_SPAN, self.config.INFER_MAX_ENTRIES)
+            data, json_logs = self._load_data(self.config.INFER_TIME_SPAN,
+                                              self.config.INFER_MAX_ENTRIES)
             if data is None:
                 time.sleep(5)
                 continue
 
-            _LOGGER.info("%d logs loaded from the last %d seconds", len(data), self.config.INFER_TIME_SPAN)
+            _LOGGER.info("%d logs loaded from the last %d seconds",
+                         len(data),
+                         self.config.INFER_TIME_SPAN)
 
             try:
                 self.w2v_model.update(data)
             except KeyError:
-                _LOGGER.error("Word2Vec model fields incompatible with current log set. Retrain model with log data from the same service")
+                _LOGGER.error("Word2Vec model fields incompatible")
+                _LOGGER.error("Retrain model with log data")
                 exit()
 
             v = self.w2v_model.one_vector(data)
 
             dist = []
 
-            dist = self.model.get_anomaly_score(v, self.config.PARALLELISM)
+            dist = self.model.get_anomaly_score(
+                v,
+                self.config.PARALLELISM)
 
             f = []
 
-            _LOGGER.info("Max anomaly score: %f" % max(dist))
+            _LOGGER.info("Max anomaly score: %f" %
+                         max(dist))
             for i in range(len(data)):
-                _LOGGER.debug("Updating entry %d - dist: %f; mean: %f" % (i, dist[i], mean))
-                s = json_logs[i]    # This needs to be more general, only works for ES incoming logs right now.
+                _LOGGER.debug("Updating entry %d - dist: %f; mean: %f" %
+                              (i, dist[i], mean))
+
+                # TODO: This needs to be more general,
+                #       only works for ES incoming logs right now.
+                s = json_logs[i]
                 s['anomaly_score'] = dist[i]
 
                 if dist[i] > threshold:
                     ANOMALY_COUNT.inc()
-                    s['anomaly'] = 1
-                    _LOGGER.warn("Anomaly found (score: %f): %s" % (dist[i], s['message']))
+                    s['predict_id'] = str(uuid.uuid4())
+                    s['anomaly'] = AnomalyEvent(
+                        s['predict_id'],
+                        s['message'], dist[i], True,
```

Comment:
From the point of understandability, maybe create an attribute `AnomalyEvent.IS_ANOMALY  = True` (or something like that) and use it instead of `True`  - so that we avoid magic constants

Reply:
By design I wanted to have Enums that have a value associated with a type. When unit testing I'll be checking for the type. I don't think I should change this to a bool.